File: ios/src/ExpoGovatarModule.swift
import ExpoModulesCore
import Govatar

public class ExpoGovatarModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoGovatar")

    AsyncFunction("getAvatar") { (username: String, gender: Int) in
      return GovatarGenerateAvatar(username, gender)
    }
  }
}

File: android/src/main/java/expo/modules/govatar/ExpoGovatarModule.kt
package expo.modules.govatar

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import java.net.URL

import govatar.Govatar

class ExpoGovatarModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoGovatar")

    AsyncFunction("getAvatar") { username: String, gender: Long ->

      return@AsyncFunction Govatar.generateAvatar(username, gender)
    }
  }
}
